### DISCUSSION TITLE
audit logs help

### DISCUSSION THREAD

**Discussion by Floffah** [discordjs/discord.js#6170] [Q&A]
ive found that in v13 when i run Guild.fetchAuditLogs() and pass the user property, neither actions that are executed by that user, or actions that target that user show up and i get an empty collection however if i omit that property and just pass the type, all of the executed and targeted audits for that user show up as well as the rest, should i create an issue? or am i doing something wrong? i have the correct intents enabled for the action and all partials enabled

**Comment by Jiralite** [comment]
> or am i doing something wrong

We can't tell without knowing your approach. Show what you've done!

**Reply by Floffah** [reply]
this is the code

```ts
class {
    async findExecutor(
        action: keyof GuildAuditLogsActions,
        id: Snowflake,
        guild: Guild,
    ) {
        const audit = await guild.fetchAuditLogs({
            type: action,
        });

        for (const [, a] of audit.entries) {
            if (
                a.action === action &&
                !!a.target &&
                "id" in a.target &&
                a.target.id === id
            ) {
                return {
                    by: `<@${a.executor?.id}> - ${a.executor?.tag} (${a.executor?.id})`,
                    timestamp: a.createdTimestamp,
                };
            }
        }

        return {
            by: "Could not fetch action executor",
            timestamp: undefined,
        };
    }
}
```

that works,
but if i add `user: id` it fails and the object returned from guild.fetchAuditLogs has an empty collection and has nothing in the object returned from .toJson() either

**Reply by Jiralite** [reply]
I've just copied and pasted your code:
- `action` was "MESSAGE_DELETE"
- The target id to check for was that of an author whose message I deleted
- I added `user: id` to `guild.fetchAuditLogs()` - my id

The code ran and I see my result. Maybe in your case, there are just no results?

How about instead just doing something simple:
```JavaScript
<Guild>.fetchAuditLogs({
  user: "id"
}).then(console.log);
```
If you see the results you expect here from the executor, then the error is in how you manipulate that data.

**Reply by Floffah** [reply]
when i pass the id as the user option i get no results but when i dont pass that option i get tonnes of results from that function

**Reply by Jiralite** [reply]
The simple code I posted above worked for me in 5e70694 which is currently the latest current commit. It doesn't get any more simple than that, but if you're not getting results still, try updating & if you still don't get any, maybe there just aren't any results?

**Reply by Floffah** [reply]
its on the latest version but there definitely is results
